Sources/GRPC/GRPCChannel/GRPCChannel.swift

```diff
@@ -19,6 +19,53 @@ import NIOSSL
 import SwiftProtobuf
 
 public protocol GRPCChannel {
+  /// Makes a gRPC call on the channel with requests and responses conforming to
+  /// `SwiftProtobuf.Message`.
+  ///
+  /// Note: this is a lower-level construct that any of `UnaryCall`, `ClientStreamingCall`,
+  /// `ServerStreamingCall` or `BidirectionalStreamingCall` and does not have an API to protect
+  /// users against protocol violations (such as sending to requests on a unary call).
+  ///
+  /// After making the `Call`, users must `invoke` the call with a callback which is invoked
+  /// for each response part (or error) received. Any call to `send(_:promise:)` prior to calling
+  /// `invoke` will fail and not be sent. Users are also responsible for closing the request stream
+  /// by sending the `.end` request part.
+  ///
+  /// - Parameters:
+  ///   - path: The path of the RPC, e.g. "/echo.Echo/get".
+  ///   - type: The type of the RPC, e.g. `.unary`.
+  ///   - callOptions: Options for the RPC.
+  ///   - interceptors: A list of interceptors to intercept the request and response stream with.
+  func makeCall<Request: SwiftProtobuf.Message, Response: SwiftProtobuf.Message>(
```

Comment:
Are we trying to wave users away from this? Do we want underscores in the names, if so?

Reply:
Not quite, they should prefer the wrapped calls but there are situations where using this is preferred (e.g. retrying interceptors) so I don't think we should include underscores.